desk/app/groups.hoon

```diff
@@ -149,6 +163,26 @@
     =+  !<(=flag:g vase)
     ga-abet:ga-invite-reject:(ga-abed:gang-core flag)
   ==
+::  +load: load next state
+++  load
+  |=  =vase
+  ^+  cor
+  =+  !<(old=versioned-state vase)
+  =?  old  ?=(%0 -.old)
+    ;;(adapted-state !<(versioned-state vase))
```

Comment:
Can you explain how this works with the previous comment?

Reply:
we `!<` the `vase` as a `versioned-state` so that we can dispatch off the old version (i.e if we need to pass cards when upgrading), and since `adapted-state` is still a `versioned-state` you can `;;` it to update it to the latest state definition as described in the previous comment.

this lets you cleanly separate the state update from any effects that need to be triggered during the update by defining the former in the `+jack` gate and the latter in `+on-load`. if you don't need to trigger any effects you just can just `;;` and be done.

Reply:
line 172 should probably just be `;;(adapted-state old)` though.

Reply:
ah ok, this is cool